src/test/scala/calc/RunTest.scala

```diff
@@ -18,7 +19,7 @@ import org.scalajs.jsenv.JSConsole
  *  result.
  */
 class RunTest {
-
+  val r = Random
```

Comment:
I guess this was related to https://github.com/sjrd/scalajsir-calc/pull/2#issuecomment-199785139 ?

I think @gzm0's point is that, because your mutable map `idToType`, your implementation will actually *accept* this program, whereas it should report a compile-time error, precisely because of the scoping issue.